libseccomp-sys/src/lib.rs

```diff
@@ -3,22 +3,199 @@
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct scmp_arg_cmp {
+    pub arg: c_uint,
+    pub op: scmp_compare,
+    pub datum_a: scmp_datum_t,
+    pub datum_b: scmp_datum_t,
+}
+
+pub const SCMP_ARCH_NATIVE: u32 = 0x0;
+pub const SCMP_ARCH_X86: u32 = 0x40000003;
+pub const SCMP_ARCH_X86_64: u32 = 0xc000003e;
+pub const SCMP_ARCH_X32: u32 = 0x4000003e;
+pub const SCMP_ARCH_ARM: u32 = 0x40000028;
+pub const SCMP_ARCH_AARCH64: u32 = 0xc00000b7;
+pub const SCMP_ARCH_MIPS: u32 = 0x8;
+pub const SCMP_ARCH_MIPS64: u32 = 0x80000008;
+pub const SCMP_ARCH_MIPS64N32: u32 = 0xa0000008;
+pub const SCMP_ARCH_MIPSEL: u32 = 0x40000008;
+pub const SCMP_ARCH_MIPSEL64: u32 = 0xc0000008;
+pub const SCMP_ARCH_MIPSEL64N32: u32 = 0xe0000008;
+pub const SCMP_ARCH_PPC: u32 = 0x14;
+pub const SCMP_ARCH_PPC64: u32 = 0x80000015;
+pub const SCMP_ARCH_PPC64LE: u32 = 0xc0000015;
+pub const SCMP_ARCH_S390: u32 = 0x16;
+pub const SCMP_ARCH_S390X: u32 = 0x80000016;
+pub const SCMP_ARCH_PARISC: u32 = 0xf;
+pub const SCMP_ARCH_PARISC64: u32 = 0x8000000f;
+pub const SCMP_ARCH_RISCV64: u32 = 0xc00000f3;
+
+pub const SCMP_ACT_MASK: u32 = 0xffff0000;
+pub const SCMP_ACT_KILL_PROCESS: u32 = 0x80000000;
+pub const SCMP_ACT_KILL_THREAD: u32 = 0x00000000;
+pub const SCMP_ACT_KILL: u32 = SCMP_ACT_KILL_THREAD;
+pub const SCMP_ACT_TRAP: u32 = 0x00030000;
+pub const SCMP_ACT_NOTIFY: u32 = 0x7fc00000;
+pub const SCMP_ACT_ERRNO_MASK: u32 = 0x00050000;
+pub const fn SCMP_ACT_ERRNO(x: u16) -> u32 {
+    SCMP_ACT_ERRNO_MASK | x as u32
 }
+pub const SCMP_ACT_TRACE_MASK: u32 = 0x7ff00000;
+pub const fn SCMP_ACT_TRACE(x: u16) -> u32 {
+    SCMP_ACT_TRACE_MASK | x as u32
+}
+pub const SCMP_ACT_LOG: u32 = 0x7ffc0000;
+pub const SCMP_ACT_ALLOW: u32 = 0x7fff0000;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct seccomp_data {
+    pub nr: c_int,
+    pub arch: u32,
+    pub instruction_pointer: u64,
+    pub args: [u64; 6],
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct seccomp_notif {
+    pub id: u64,
+    pub pid: u32,
+    pub flags: u32,
+    pub data: seccomp_data,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct seccomp_notif_resp {
+    pub id: u64,
+    pub val: i64,
+    pub error: i32,
+    pub flags: u32,
+}
+
+#[link(name = "seccomp")]
+extern "C" {
+    pub fn seccomp_version() -> *const scmp_version;
+    pub fn seccomp_api_get() -> c_uint;
+    pub fn seccomp_api_set(level: c_uint) -> c_int;
+    pub fn seccomp_init(def_action: u32) -> scmp_filter_ctx;
+    pub fn seccomp_reset(ctx: scmp_filter_ctx, def_action: u32) -> c_int;
+    pub fn seccomp_release(ctx: scmp_filter_ctx);
+    pub fn seccomp_merge(ctx_dst: scmp_filter_ctx, ctx_src: scmp_filter_ctx) -> c_int;
+    pub fn seccomp_arch_resolve_name(arch_name: *const c_char) -> u32;
+    pub fn seccomp_arch_native() -> u32;
+    pub fn seccomp_arch_exist(
+        ctx: /* FIXME: *const c_void */ scmp_filter_ctx,
```

Comment:
Do we need to make `scmp_filter_ctx` `const`?

Reply:
I agree with we just stick to `*mut c_void`/scmp_filter_ctx` because `scmp_filter_ctx` type is easy to understand.

Reply:
Hmm, but it would be better to use `type const_scmp_filter_ctx = *const c_void` to make `libseccomp-sys` more accurate.